utils/prod/update_config.py

```diff
@@ -83,33 +105,39 @@ def update_configs(mode: str = 'prod') -> None:
             print(f'Config file not found for {kit_name}. Skipping.')
             continue
 
-        with open(config_path, 'r') as file:
+        with open(config_path, 'r', encoding='utf-8') as file:
             config: Dict[str, Any] = yaml.safe_load(file)
 
         if mode == 'prod':
             update_config_prod(config)
         else:
             update_embedding_model(config)
 
-        with open(config_path, 'w') as file:
-            yaml.dump(config, file)
+        with open(config_path, 'w', encoding='utf-8') as file:
+            yaml.dump(config, file, default_flow_style=False)
 
         print(f'Updated config for {kit_name} in {mode} mode.')
 
     if mode == 'prod':
-        update_makefile(root_dir)
+        update_makefile(root_dir, port)
 
 
-if __name__ == '__main__':
-    import sys
+def main() -> None:
+    """Main function to parse arguments and run the update process."""
+    parser = argparse.ArgumentParser(description='Update configurations for AI Starter Kits.')
+    parser.add_argument(
+        'mode',
+        nargs='?',
+        default='prod',
+        choices=['prod', 'test'],
```

Comment:
Why choosing the name `test` here?
Is test mode the standard (as opposed to prod mode)? Why not dev mode?

Reply:
We used 'test' as this mode is for running our global test suite. Comment added